### ISSUE TITLE
Style/HashEachMethods doesn't detect &blocks

### ISSUE THREAD

**Issue by vlad-pisanov** [rubocop/rubocop#11139]
Currently, `Style/HashEachMethods` detects regular blocks and numblocks, but ignores &blocks:

```ruby
h.keys.each { ... }
# Style/HashEachMethods suggests: h.each_key { ... } ✔️

h.keys.each(&:foo)
# Style/HashEachMethods ignores ❌
```

Ideally, it should recommend:
```ruby
h.each_key(&:foo)
```
